src/common/net.ts

```diff
@@ -5,16 +5,31 @@
  */
 
 import * as net from "net";
+import { Promise as bluebird } from "bluebird";
 
-const isPortAvailable = (port: number, host: string): Promise<void> => {
+const isPortAvailable = (port: number, host: string): Promise<void> =>
+  Promise.resolve(bluebird.each([host, "127.0.0.1", "localhost", "0.0.0.0"], h =>
```

Comment:
Is this the right set of addresses? 0.0.0.0 means all addresses on the local machine, which might not always be right (and it’s also possible that there’d be cases where we don’t want it listening on the loopback, too?)

Reply:
I ummed and ah'd about it. Doing some testing on my Mac, I found that even when I listened for a connection on the loopback, the port check would pass but the mock service would fail (with the EADDRINUSE. I'm not sure if there is something in the mock server that doesn't like it.

I felt given that it's better to fail fast (and eagerly) than otherwise, and that given it's a testing tool the environment should be more flexible.

Unless I'm blind to a class of errors?